php/Exchange.php

```diff
@@ -1517,6 +1517,29 @@ public function parseOrderBook($orderbook, $timestamp = null, $bids_key = 'bids'
 
     public function parse_balance($balance) {
         $currencies = array_keys($this->omit($balance, 'info'));
+
+        $balance['free'] = array();
+        $balance['used'] = array();
+        $balance['total'] = array();
+
+        foreach ($currencies as $currency) {
+            if ($currencies[$currency]['total'] === 0.0) {
+                if ($currencies[$currency]['free'] !== null && $currencies[$currency]['used'] !== null) {
+                    $currencies[$currency]['total'] = static::sum($currencies[$currency]['free'], $currencies[$currency]['used']);
+                }
+            }
+            if ($currencies[$currency]['used'] === 0.0) {
+                if ($currencies[$currency]['total'] !== null && $currencies[$currency]['free'] !== null) {
+                    $currencies[$currency]['used'] = static::sum($currencies[$currency]['total'], -$currencies[$currency]['free']);
+                }
+            }
+            if ($currencies[$currency]['free'] === 0.0) {
+                if ($currencies[$currency]['total'] !== null && $currencies[$currency]['used'] !== null) {
```

Comment:
Have you tested this? Because it doesn't really work in PHP this way, we should use `isset()` or `array_key_exists` here.